### PR TITLE
Linter

### PR DESCRIPTION
**Pull request by georgiybykov**
- Rubocop
- GitHub Actions for Rubocop
- Add unique index for users email
- Minor fixes

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -4,50 +4,113 @@ require:
   - rubocop-rspec
 
 AllCops:
-  TargetRubyVersion: 2.7
+  TargetRubyVersion: 2.7.2
```
Comment (faraquet) [inline .rubocop.yml]: Do we really need such minor version?

File: app/controllers/pdf_controller.rb [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+    browser.go_to("file:///#{Rails.root.join("tmp/#{filename}.html")}")
     pdf_options = { path: "tmp/#{filename}.pdf" }
 
-    pdf_options[:landscape] = ActiveRecord::Type::Boolean.new.deserialize(params[:landscape]) if params[:landscape].present?
+    if params[:landscape].present?
+      pdf_options[:landscape] =
+        ActiveRecord::Type::Boolean.new.deserialize(params[:landscape])
+    end
+
     pdf_options[:scale] = ActiveRecord::Type::Float.new.deserialize(params[:scale]) if params[:scale].present?
```
Comment (faraquet) [inline app/controllers/pdf_controller.rb]: Let's change it to uniformity:

```
 if ...
	...
end
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,15 +1,19 @@
+# frozen_string_literal: true
+
 class UsersController < ApplicationController
   skip_before_action :authenticate_request
 
-  before_action only: :create, unless: -> { ActiveRecord::Type::Boolean.new.deserialize(ENV["SELF_REGISTRATION_AVAILABLE"]) } do
-    render json: { error: "Self registration is not available" }, status: :unprocessable_entity
+  before_action only: :create, unless: lambda {
```
Comment (faraquet) [inline app/controllers/users_controller.rb]: what's the reason to change `->` to `lambda` word?

File: config/application.rb [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
+require 'action_cable/engine'
 # require "sprockets/railtie"
-require "rails/test_unit/railtie"
+require 'rails/test_unit/railtie'
 
 # Require the gems listed in Gemfile, including any gems
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-module Fernatra
```
Comment (faraquet) [inline config/application.rb:22]: :+1:

File: db/migrate/20220202212157_add_uniq_index_for_email_to_users.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddUniqIndexForEmailToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_index :users, :email, unique: true
```
Comment (faraquet) [inline db/migrate/20220202212157_add_uniq_index_for_email_to_users.rb:5]: :+1:

**Review by faraquet — APPROVED** [review]
Let's also return `.keep` files and go ahead
:rocket:

File: .rubocop.yml (hunk 1 above)
Comment (georgiybykov) [reply .rubocop.yml]: I do not think so. Les't try to get rid of it.

File: app/controllers/pdf_controller.rb (hunk 2 above)
Comment (georgiybykov) [reply app/controllers/pdf_controller.rb]: Okay, however I have to disable the Rubocop `Style/IfUnlessModifier` for this line. Please check the [docs](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/IfUnlessModifier) for this rule to decide whether or not to use this check for the project.

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (georgiybykov) [reply app/controllers/users_controller.rb]: We can control this by the rule `Style/Lambda` cutomization:

```
Style/Lambda: Use the lambda method for multiline lambdas.
```
I have changed it to:
```
Style/Lambda:
  EnforcedStyle: literal
```
So, we will use `->` for any case from now.

**Comment by georgiybykov** [comment]
@faraquet, I have commited minor changes according to the comments. Take a look, please:)

**Pull request merged**